### DISCUSSION TITLE
Query string support

### DISCUSSION THREAD

**Discussion by sandulat** [sandulat/routes-gen#16] [Ideas]
It would be great if the `route` function allowed to optionally add query string parameters to the route.

### Example:

Usage: `route("/products", { sort: "asc" })`
Output: `/products?sort=asc`

Usage: `route("/products/:productId/comments", { productId: "123"}, { sort: "asc" })`
Output: `/products/123/comments?sort=asc`.

> This discussion is a follow-up on https://github.com/sandulat/routes-gen/issues/11.

**Comment by IgnisDa** [comment]
@sandulat any updates on this?

**Comment by robyhuzwandar** [comment]
what is the status of this topic,

can we go with this example ?

Usage: `route("/products", { sort: "asc" })`
Output: `/products?sort=asc`

**Comment by sandulat** [comment]
@robyhuzwandar @IgnisDa I'm still not sure if this should belong to this package since it was not meant to be an universal tool to work with URLs, but rather add typings for your route list and their params **only**.

It can be easily solved in multiple ways:
```js
`${route("/products")}?sort=asc`
```
```js
`${route("/products")}?${new URLSearchParams({ sort: 'asc' })}`
```

IMHO there is literally no point in adding this to `routes-gen` since query params **cannot** be type safe anyway, so you can just use basic string interpolation or native web apis for that.

**Reply by sandulat** [reply]
I'm also not a big fan of functions overloading (which will have to be used in order to achieve that):
1. `route("/products", { sort: "asc" })` - **second argument** is for **query params**.
2. `route("/products/:productId", { productId: 1 }, { sort: "asc" })` - third argument is for query params, and the **second one** is now for **route params**.

This type of development is IMHO error prone and I'd want people to expect the same behaviour when providing arguments.
